source/core/ut_suite_manager.pkb

```diff
@@ -327,5 +339,166 @@ create or replace package body ut_suite_manager is
+      l_paths ut_varchar2_list := ut_varchar2_list();
+    begin
+      l_paths.extend(a_paths.count);
+      for i in 1 .. a_paths.count loop
+        l_paths(i) := trim(lower(a_paths(i)));
+      end loop;
+      l_paths := set(l_paths);
+      return l_paths;
+    end clean_paths;
+  
+    procedure set_skipped_flag(a_suite in out nocopy ut_test_object, a_path varchar2) is
+      l_root      constant varchar2(32767) := regexp_substr(a_path, '\w+');
+      l_rest_path constant varchar2(32767) := regexp_substr(a_path, '\.(.+)', subexpression => 1);
+      l_item  ut_test_object;
+      l_items ut_objects_list := ut_objects_list();
+      l_object_name varchar2(32767);
+      
+    begin
+      a_suite.set_ignore_flag(false);
+    
+      if a_path is not null then
+      
+        for i in 1 .. a_suite.items.count loop
+        
+          l_item := treat(a_suite.items(i) as ut_test_object);
+        
+          l_object_name := l_item.object_name;
+          --l_object_name := regexp_substr(l_object_name,'\w+$'); -- temporary fix. seems like suite have suitepath in object_name
+          if regexp_like(l_object_name, l_root, modifier => 'i') then
+            
+            set_skipped_flag(l_item, l_rest_path);          
+            l_items.extend;
+            l_items(l_items.count) := l_item;
+
+          end if;
+          
+        end loop;
+        
+        a_suite.items := l_items;
+      
+        if l_items.count = 0 then--not l_found then
+          raise_application_error(-20203, 'Suite note found');
+        end if;
+      end if;
+    end set_skipped_flag;
+  begin
+    l_paths := clean_paths(a_paths);
+  
+    validate_paths(l_paths);
+  
+    -- current implementation operates only on a single path
+    -- to be improved later
+    for i in 1 .. l_paths.count loop
+      l_path   := l_paths(i);
+      l_schema := regexp_substr(l_path, '^(\w+)(\.|:|$)', 1, 1, null, 1);
+    
+      l_schema_suites := get_schema_suites(upper(l_schema));
+    
+      if regexp_like(l_path, '^\w+$') then
+        -- run whole schema
+        l_index := l_schema_suites.first;
+        while l_index is not null loop
+          l_objects_to_run.extend;
+          l_objects_to_run(l_objects_to_run.count) := l_schema_suites(l_index);
+          l_index := l_schema_suites.next(l_index);
+        end loop;
+      else
+        -- convert SCHEMA.PACKAGE.PROCEDURE syntax to fully qualified path
+        if regexp_like(l_path, '^\w+(\.\w+){1,2}$') then
+          declare
+            l_temp_suite     ut_test_suite;
+            l_package_name   varchar2(4000);
+            l_procedure_name varchar2(4000);
+          begin
+            l_package_name   := regexp_substr(l_path, '^\w+\.(\w+)(\.(\w+))?$', subexpression => 1);
+            l_procedure_name := regexp_substr(l_path, '^\w+\.(\w+)(\.(\w+))?$', subexpression => 3);
+          
+            l_temp_suite := config_package(l_schema, l_package_name);
```

Comment:
Then we will have to work through the whole hierarcy looking for `object_name`... but maybe you are right. i'll come back to it today a bit later.